pkg/executor/insert.go

```diff
@@ -326,6 +326,23 @@ func optimizeDupKeyCheckForNormalInsert(vars *variable.SessionVars, txn kv.Trans
 	return table.DupKeyCheckInPlace
 }
 
+// getPessimisticLazyCheckMode returns the lazy check mode for pessimistic txn.
+func getPessimisticLazyCheckMode(vars *variable.SessionVars) table.PessimisticLazyDupKeyCheckMode {
```

Comment:
`IsPessimistic` seems not a factor to select `DupKeyCheckInPrewrite` or `DupKeyCheckInAcquireLock`, because both of these two enums are only used to pessimistic mode.